### PR TITLE
Restart batch with active deployment

### PR DESCRIPTION
**Pull request by satr**

### CODE REVIEW

**Review by nilsgstrabo — CHANGES_REQUESTED** [review]
As discussed, reuse secret from source RadixBatch.
Added a few minor suggestions

File: api/v1/batches/batch_handler.go [hunk 1]
```diff
@@ -29,21 +29,23 @@ type batchHandler struct {
...
+	MaintainHistoryLimit(context.Context) error
 	// DeleteBatch Delete a batch
-	DeleteBatch(string) error
+	DeleteBatch(context.Context, string) error
 	// StopBatch Stop a batch
-	StopBatch(string) error
+	StopBatch(context.Context, string) error
 	// StopBatchJob Stop a batch job
-	StopBatchJob(string, string) error
+	StopBatchJob(context.Context, string, string) error
```
[the remark this answers is not in the document]
Comment (satr) [reply api/v1/batches/batch_handler.go]: fixed

File: api/v1/jobs/job_handler.go [hunk 2]
```diff
@@ -30,17 +30,19 @@ type jobHandler struct {
...
+	CopyJob(context.Context, string, string) (*modelsv1.JobStatus, error)
 	// MaintainHistoryLimit Delete outdated jobs
-	MaintainHistoryLimit() error
+	MaintainHistoryLimit(context.Context) error
 	// DeleteJob Delete a job
-	DeleteJob(string) error
+	DeleteJob(context.Context, string) error
 	// StopJob Stop a job
-	StopJob(string) error
+	StopJob(context.Context, string) error
```
[the remark this answers is not in the document]
Comment (satr) [reply api/v1/jobs/job_handler.go]: fixed

File: api/v2/handler.go [hunk 3]
```diff
@@ -489,18 +510,121 @@ func (h *handler) createRadixBatch(namespace, appName, radixDeploymentName strin
...
+		job := sourceJob.DeepCopy()
+		job.Name = createJobName()
+		sourceJobNameToNewNameMap[sourceJob.Name] = job.Name
+		radixBatch.Spec.Jobs = append(radixBatch.Spec.Jobs, *job)
+	}
+	return sourceJobNameToNewNameMap
+}
+
+func (h *handler) copyPayloadSecrets(ctx context.Context, sourceRadixBatch *radixv1.RadixBatch, namespace, appName, radixComponentName, batchName string, sourceJobNameToNewNameMap map[string]string, sourceJobName string) (map[string]string, error) {
+	sourceBatchSecrets, err := h.GetSecretsForRadixBatch(sourceRadixBatch.GetName())
```
[the remark this answers is not in the document]
Comment (satr) [reply api/v2/handler.go]: fixed

**Review by nilsgstrabo — APPROVED** [review]

**Pull request merged**